## Running a chi-squared test, t-test, and a correlation / regression on SAT Results data and interpreting the results.

In [10]:
import pandas as pd # importing pandas library for use of pandas functions such as pd.melt and declaring it as 'pd' which will become easy to refer instead of writing pandas repeatedly 
import numpy as np

***Importing data of SAT_Result of year 2010***

In [11]:
    fp = 'C:/Users/Ashwini/Desktop/Intro to DS/SAT_Result_2010.csv' # Assigning path of file to variable 'fp', to make use of it to read the file
    SAT_Result_2010 = pd.read_csv(fp) # read csv file and save the data in dataframe SAT_Result_2010
    SAT_Result_2010 # display first 20 records of SAT_Result_2010 dataframe

DBN                                        School Name  \
0    01M292     Henry Street School for International Studies    
1    01M448               University Neighborhood High School    
2    01M450                   East Side Community High School    
3    01M458                      SATELLITE ACADEMY FORSYTH ST    
4    01M509                                  CMSP HIGH SCHOOL    
5    01M515           Lower East Side Preparatory High School    
6    01M539       New Explorations into Sci, Tech and Math HS    
7    01M650                              CASCADES HIGH SCHOOL    
8    01M696                    BARD HIGH SCHOOL EARLY COLLEGE    
9    02M047                       AMERICAN SIGN LANG ENG DUAL    
10   02M288                       FOOD AND FNANCE HIGH SCHOOL    
11   02M294                     HIGH SCHOOL FOR HIST AND COMM    
12   02M296             High School of Hospitality Management    
13   02M298                                  PACE HIGH SCHOOL    
14   02M300  Urban Assembly School of Design and Construction    
15   02M303                         The Facing History School    
16   02M305      Urban Assembly Academy of Government and Law    
17   02M308                      LOWER MANHATTAN ARTS ACADEMY    
18   02M313                          The James Baldwin School    
19   02M316  Urban Assembly School of Business for Young Wo...   
20   02M374                         GRAMERCY ARTS HIGH SCHOOL    
21   02M376                                       NYC iSchool    
22   02M400                     HIGH SCHOOL ENVRNMNTL STUDIES    
23   02M407             Institute for Collaborative Education    
24   02M408                Professional Performng Arts School    
25   02M411                 Baruch College Campus High School    
26   02M412    New York City Laboratory School Collab Studies    
27   02M413                              SCHOOL OF THE FUTURE    
28   02M414                       NEW YORK CITY MUSEUM SCHOOL    
29   02M416                     ELEANOR ROOSEVELT HIGH SCHOOL    
..      ...                                                ...   
430  30Q555                             NEWCOMERS HIGH SCHOOL    
431  30Q575                       ACADEMY OF AMERICAN STUDIES    
432  30Q580                     BACCALAUREATE SCH GLOBAL EDUC    
433  31R064       Gaynor McCown Expeditionary Learning School    
434  31R080                         MICHAEL J PETRIDES SCHOOL    
435  31R440                              NEW DORP HIGH SCHOOL    
436  31R445                         PORT RICHMOND HIGH SCHOOL    
437  31R450                                CURTIS HIGH SCHOOL    
438  31R455                           TOTTENVILLE HIGH SCHOOL    
439  31R460                        SUSAN E WAGNER HIGH SCHOOL    
440  31R470                               CONCORD HIGH SCHOOL    
441  31R600                     RALPH MCKEE VOC-TECH HIGH SCH    
442  31R605               STATEN ISLAND TECHNICAL HIGH SCHOOL    
443  32K403               ACADEMY OF ENVIRONMENTAL LEADERSHIP    
444  32K480                              Bushwick High School    
445  32K545     EBC High School for Public Service - Bushwick    
446  32K549                    BUSHWICK HS FOR SOCIAL JUSTICE    
447  32K552                         Academy of Urban Planning    
448  32K554                     ALL CITY LEADERSHIP SECONDARY    
449  32K556   Bushwick Leaders High School for Academic Excel    
450  72X470                           South Bronx High School    
451  75K371                        LILLIAN L RASHKIS HS P371K    
452  75M035                      MANHATTAN HIGH SCHOOL - P35M    
453  75Q004                                P4Q SCHOOL (@ 179)    
454  75Q256                 P256 QUEENS SCHOOL @ SAINT MARY'S    
455  75R025                        South Richmond High School    
456  75X012                      PS12X LEWIS AND CLARK SCHOOL    
457  75X754  P754 X - Jeffrey M. Rapport School for Career ...   
458  76K460                              John Ja

In [12]:
column_names = {'DBN': 'DB_Number_2010', 'School Name': 'SCHOOL_2010', 'Number of Test Takers': 'Applicants_2010',
                'Critical Reading Mean': 'CR_Average_Score_2010', 'Mathematics Mean': 'Math_Average_Score_2010', 
                'Writing Mean': 'Writing_Average_Score_2010'} # Creating dictionary column_names to store key-value pairs for renaming column names 

SAT_Result_2010 = (SAT_Result_2010.rename(columns=column_names)# renaming column names of 'SAT_Result_2010' dataframe
    .dropna(thresh=4)# dropping all rows from dataframe with more than 4 NaN values
    [['DB_Number_2010','SCHOOL_2010','Applicants_2010','CR_Average_Score_2010','Math_Average_Score_2010','Writing_Average_Score_2010']]# columns to be displayed in output table
    .set_index('DB_Number_2010', append=True) # setting the index of 'SAT_Result_2010' dataframe using existing column and append is set to true to append column to an existing index
    .rename_axis(["Serial Number", "DB_Number_2010"])  # altering an index of Serial Number and DB_Number
    .sort_index())  # sorting all dataframe indexes



In [13]:
SAT_Result_2010.replace('s',np.NaN) # replace 's' values of all columns with NaN value using replace function
SAT_Result_2010["Applicants_2010"]=pd.to_numeric(SAT_Result_2010["Applicants_2010"], errors="coerce") # converting datatype of Applicant_2010 column to Numeric
SAT_Result_2010["CR_Average_Score_2010"]=pd.to_numeric(SAT_Result_2010["CR_Average_Score_2010"], errors="coerce") # converting datatype of CR_Average_Score_2010 column to Numeric
SAT_Result_2010["Math_Average_Score_2010"]=pd.to_numeric(SAT_Result_2010["Math_Average_Score_2010"], errors="coerce") # converting datatype of Math_Average_Score_2010 column to Numeric
SAT_Result_2010["Writing_Average_Score_2010"]=pd.to_numeric(SAT_Result_2010["Writing_Average_Score_2010"], errors="coerce") # converting datatype of Writing_Average_Score_2010 column to Numeric

In [14]:
SAT_Result_2010["Applicants_2010"].fillna(int(SAT_Result_2010["Applicants_2010"].mean()), inplace=True) # replace NaN values of Applicants column with mean value of same column
SAT_Result_2010["CR_Average_Score_2010"].fillna(int(SAT_Result_2010["CR_Average_Score_2010"].mean()), inplace=True)# replace NaN values of Applicants column with mean value of same columnce=True) # replace NaN values of CR_Average_Score column with mean value of same column
SAT_Result_2010["Math_Average_Score_2010"].fillna(int(SAT_Result_2010["Math_Average_Score_2010"].mean()), inplace=True)# replace NaN values of Math_Average_Score column with mean value of same column
SAT_Result_2010["Writing_Average_Score_2010"].fillna(int(SAT_Result_2010["Writing_Average_Score_2010"].mean()), inplace=True)# replace NaN values of Writing_Average_Score column with mean value of same column

SAT_Result_2010 # displaying entire dataset

SCHOOL_2010  \
Serial Number DB_Number_2010                                                      
0             01M292             Henry Street School for International Studies    
1             01M448                       University Neighborhood High School    
2             01M450                           East Side Community High School    
3             01M458                              SATELLITE ACADEMY FORSYTH ST    
4             01M509                                          CMSP HIGH SCHOOL    
5             01M515                   Lower East Side Preparatory High School    
6             01M539               New Explorations into Sci, Tech and Math HS    
7             01M650                                      CASCADES HIGH SCHOOL    
8             01M696                            BARD HIGH SCHOOL EARLY COLLEGE    
9             02M047                               AMERICAN SIGN LANG ENG DUAL    
10            02M288                               FOOD AND FNANCE HIGH SCHOOL    
11            02M294                             HIGH SCHOOL FOR HIST AND COMM    
12            02M296                     High School of Hospitality Management    
13            02M298                                          PACE HIGH SCHOOL    
14            02M300          Urban Assembly School of Design and Construction    
15            02M303                                 The Facing History School    
16            02M305              Urban Assembly Academy of Government and Law    
17            02M308                              LOWER MANHATTAN ARTS ACADEMY    
18            02M313                                  The James Baldwin School    
19            02M316          Urban Assembly School of Business for Young Wo...   
20            02M374                                 GRAMERCY ARTS HIGH SCHOOL    
21            02M376                                               NYC iSchool    
22            02M400                             HIGH SCHOOL ENVRNMNTL STUDIES    
23            02M407                     Institute for Collaborative Education    
24            02M408                        Professional Performng Arts School    
25            02M411                         Baruch College Campus High School    
26            02M412            New York City Laboratory School Collab Studies    
27            02M413                                      SCHOOL OF THE FUTURE    
28            02M414                               NEW YORK CITY MUSEUM SCHOOL    
29            02M416                             ELEANOR ROOSEVELT HIGH SCHOOL    
...                                                                         ...   
430           30Q555                                     NEWCOMERS HIGH SCHOOL    
431           30Q575                               ACADEMY OF AMERICAN STUDIES    
432           30Q580                             BACCALAUREATE SCH GLOBAL EDUC    
433           31R064               Gaynor McCown Expeditionary Learning School    
434           31R080                                 MICHAEL J PETRIDES SCHOOL    
435           31R440                                      NEW DORP HIGH SCHOOL    
436           31R445                                 PORT RICHMOND HIGH SCHOOL    
437           31R450                                        CURTIS HIGH SCHOOL    
438           31R455                                   TOTTENVILLE HIGH SCHOOL    
439           31R460                                SUSAN E WAGNER HIGH SCHOOL    
440           31R470                                       CONCORD HIGH SCHOOL    
441           31R600                             RALPH MCKEE VOC-TECH HIGH SCH    
442           31R605                       STATEN ISLAND TECHNICAL HIGH SCHOOL    
443           32K403                       ACADEMY OF ENVIRONMENTAL LEADERSHIP    
444           32K480                                      Bushwick High School    
445           32K545             EBC High School for Public Service - Bushwick    
4

***Importing data of SAT_Result of year 2012***

In [15]:
    fp = 'C:/Users/Ashwini/Desktop/Intro to DS/SAT_Results.csv' # Assigning path of file to variable 'fp', to make use of it to read the file
    SAT_Result = pd.read_csv(fp) # read csv file and save the data in dataframe SAT_Result
    SAT_Result # display first 20 records of SAT_Result dataframe 
    

DBN                                        SCHOOL NAME  \
0    01M292      HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1    01M448                UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2    01M450                         EAST SIDE COMMUNITY SCHOOL   
3    01M458                          FORSYTH SATELLITE ACADEMY   
4    01M509                            MARTA VALLE HIGH SCHOOL   
5    01M515            LOWER EAST SIDE PREPARATORY HIGH SCHOOL   
6    01M539  NEW EXPLORATIONS INTO SCIENCE, TECHNOLOGY AND ...   
7    01M650                               CASCADES HIGH SCHOOL   
8    01M696                     BARD HIGH SCHOOL EARLY COLLEGE   
9    02M047  47 THE AMERICAN SIGN LANGUAGE AND ENGLISH SECO...   
10   02M288                       FOOD AND FINANCE HIGH SCHOOL   
11   02M294                               ESSEX STREET ACADEMY   
12   02M296              HIGH SCHOOL OF HOSPITALITY MANAGEMENT   
13   02M298                                   PACE HIGH SCHOOL   
14   02M300  URBAN ASSEMBLY SCHOOL OF DESIGN AND CONSTRUCTI...   
15   02M303                         FACING HISTORY SCHOOL, THE   
16   02M305  URBAN ASSEMBLY ACADEMY OF GOVERNMENT AND LAW, THE   
17   02M308                       LOWER MANHATTAN ARTS ACADEMY   
18   02M313  JAMES BALDWIN SCHOOL, THE: A SCHOOL FOR EXPEDI...   
19   02M316  URBAN ASSEMBLY SCHOOL OF BUSINESS FOR YOUNG WO...   
20   02M374                          GRAMERCY ARTS HIGH SCHOOL   
21   02M376                                        NYC ISCHOOL   
22   02M392                         MANHATTAN BUSINESS ACADEMY   
23   02M393                          BUSINESS OF SPORTS SCHOOL   
24   02M394                           EMMA LAZARUS HIGH SCHOOL   
25   02M399         THE HIGH SCHOOL FOR LANGUAGE AND DIPLOMACY   
26   02M400              HIGH SCHOOL FOR ENVIRONMENTAL STUDIES   
27   02M407              INSTITUTE FOR COLLABORATIVE EDUCATION   
28   02M408           PROFESSIONAL PERFORMING ARTS HIGH SCHOOL   
29   02M411                  BARUCH COLLEGE CAMPUS HIGH SCHOOL   
..      ...                                                ...   
448  30Q580          BACCALAUREATE SCHOOL FOR GLOBAL EDUCATION   
449  31R047          CSI HIGH SCHOOL FOR INTERNATIONAL STUDIES   
450  31R064        GAYNOR MCCOWN EXPEDITIONARY LEARNING SCHOOL   
451  31R080                     THE MICHAEL J. PETRIDES SCHOOL   
452  31R440                               NEW DORP HIGH SCHOOL   
453  31R445                          PORT RICHMOND HIGH SCHOOL   
454  31R450                                 CURTIS HIGH SCHOOL   
455  31R455                            TOTTENVILLE HIGH SCHOOL   
456  31R460                        SUSAN E. WAGNER HIGH SCHOOL   
457  31R470                                CONCORD HIGH SCHOOL   
458  31R600  RALPH R. MCKEE CAREER AND TECHNICAL EDUCATION ...   
459  31R605                STATEN ISLAND TECHNICAL HIGH SCHOOL   
460  31R607    YOUNG ADULT BOROUGH CENTER AT STATEN IS TECH HS   
461  32K403               ACADEMY FOR ENVIRONMENTAL LEADERSHIP   
462  32K545        EBC HIGH SCHOOL FOR PUBLIC SERVICE–BUSHWICK   
463  32K549                 BUSHWICK SCHOOL FOR SOCIAL JUSTICE   
464  32K552                          ACADEMY OF URBAN PLANNING   
465  32K554               ALL CITY LEADERSHIP SECONDARY SCHOOL   
466  32K556  BUSHWICK LEADERS HIGH SCHOOL FOR ACADEMIC EXCE...   
467  32K564                     BUSHWICK COMMUNITY HIGH SCHOOL   
468  75K371                      P.S. 371 s LILLIAN L. RASHKIS   
469  75M035                                           P.S. 035   
470  75Q256                                          P.S. Q256   
471  75Q811                                          P.S. Q811   
472  75R025            SOUTH RICHMOND HIGH SCHOOL I.S./P.S. 25   
473  75X012                   P.S. X012 LEWIS AND CLARK SCHOOL   
474  75X754            J. M. RAPPORT SCHOOL CAREER DEVELOPMENT   
475  79M645         SCHOOL FOR COOPERATIVE TECHNICAL EDUCATION   
476  79Q950                                GED P

In [16]:
column_names = {'DBN': 'DB_Number', 'SCHOOL NAME': 'SCHOOL', 'Num of SAT Test Takers': 'Applicants',
                'SAT Critical Reading Avg. Score': 'CR_Average_Score', 'SAT Math Avg. Score': 'Math_Average_Score', 
                'SAT Writing Avg. Score': 'Writing_Average_Score'} # Creating dictionary column_names to store key-value pairs for renaming column names 

SAT_Result = (SAT_Result.rename(columns=column_names)# renaming column names of 'SAT_Result' dataframe
    .dropna(thresh=4)# dropping all rows from dataframe with more than 4 NaN values
    [['DB_Number','SCHOOL','Applicants','CR_Average_Score','Math_Average_Score','Writing_Average_Score']]# columns to be displayed in output table
    .set_index('DB_Number', append=True) # setting the index of 'SAT_Result' dataframe using existing column and append is set to true to append column to an existing index
    .rename_axis(["Serial Number", "DB_Number"])  # altering an index of Serial Number and DB_Number
    .sort_index())  # sorting all dataframe indexes


In [17]:
SAT_Result.replace('s',np.NaN) # replace 's' values of all columns with NaN value using replace function
SAT_Result["Applicants"]=pd.to_numeric(SAT_Result["Applicants"], errors="coerce") # converting datatype of Applicant column to Numeric
SAT_Result["CR_Average_Score"]=pd.to_numeric(SAT_Result["CR_Average_Score"], errors="coerce") # converting datatype of CR_Average_Score column to Numeric
SAT_Result["Math_Average_Score"]=pd.to_numeric(SAT_Result["Math_Average_Score"], errors="coerce") # converting datatype of Math_Average_Score column to Numeric
SAT_Result["Writing_Average_Score"]=pd.to_numeric(SAT_Result["Writing_Average_Score"], errors="coerce") # converting datatype of Writing_Average_Score column to Numeric

In [18]:
SAT_Result["Applicants"].fillna(int(SAT_Result["Applicants"].mean()), inplace=True) # replace NaN values of Applicants column with mean value of same column
SAT_Result["CR_Average_Score"].fillna(int(SAT_Result["CR_Average_Score"].mean()), inplace=True)# replace NaN values of Applicants column with mean value of same columnce=True) # replace NaN values of CR_Average_Score column with mean value of same column
SAT_Result["Math_Average_Score"].fillna(int(SAT_Result["Math_Average_Score"].mean()), inplace=True)# replace NaN values of Math_Average_Score column with mean value of same column
SAT_Result["Writing_Average_Score"].fillna(int(SAT_Result["Writing_Average_Score"].mean()), inplace=True)# replace NaN values of Writing_Average_Score column with mean value of same column

SAT_Result # displaying entire dataset

SCHOOL  \
Serial Number DB_Number                                                      
0             01M292         HENRY STREET SCHOOL FOR INTERNATIONAL STUDIES   
1             01M448                   UNIVERSITY NEIGHBORHOOD HIGH SCHOOL   
2             01M450                            EAST SIDE COMMUNITY SCHOOL   
3             01M458                             FORSYTH SATELLITE ACADEMY   
4             01M509                               MARTA VALLE HIGH SCHOOL   
5             01M515               LOWER EAST SIDE PREPARATORY HIGH SCHOOL   
6             01M539     NEW EXPLORATIONS INTO SCIENCE, TECHNOLOGY AND ...   
7             01M650                                  CASCADES HIGH SCHOOL   
8             01M696                        BARD HIGH SCHOOL EARLY COLLEGE   
9             02M047     47 THE AMERICAN SIGN LANGUAGE AND ENGLISH SECO...   
10            02M288                          FOOD AND FINANCE HIGH SCHOOL   
11            02M294                                  ESSEX STREET ACADEMY   
12            02M296                 HIGH SCHOOL OF HOSPITALITY MANAGEMENT   
13            02M298                                      PACE HIGH SCHOOL   
14            02M300     URBAN ASSEMBLY SCHOOL OF DESIGN AND CONSTRUCTI...   
15            02M303                            FACING HISTORY SCHOOL, THE   
16            02M305     URBAN ASSEMBLY ACADEMY OF GOVERNMENT AND LAW, THE   
17            02M308                          LOWER MANHATTAN ARTS ACADEMY   
18            02M313     JAMES BALDWIN SCHOOL, THE: A SCHOOL FOR EXPEDI...   
19            02M316     URBAN ASSEMBLY SCHOOL OF BUSINESS FOR YOUNG WO...   
20            02M374                             GRAMERCY ARTS HIGH SCHOOL   
21            02M376                                           NYC ISCHOOL   
22            02M392                            MANHATTAN BUSINESS ACADEMY   
23            02M393                             BUSINESS OF SPORTS SCHOOL   
24            02M394                              EMMA LAZARUS HIGH SCHOOL   
25            02M399            THE HIGH SCHOOL FOR LANGUAGE AND DIPLOMACY   
26            02M400                 HIGH SCHOOL FOR ENVIRONMENTAL STUDIES   
27            02M407                 INSTITUTE FOR COLLABORATIVE EDUCATION   
28            02M408              PROFESSIONAL PERFORMING ARTS HIGH SCHOOL   
29            02M411                     BARUCH COLLEGE CAMPUS HIGH SCHOOL   
...                                                                    ...   
448           30Q580             BACCALAUREATE SCHOOL FOR GLOBAL EDUCATION   
449           31R047             CSI HIGH SCHOOL FOR INTERNATIONAL STUDIES   
450           31R064           GAYNOR MCCOWN EXPEDITIONARY LEARNING SCHOOL   
451           31R080                        THE MICHAEL J. PETRIDES SCHOOL   
452           31R440                                  NEW DORP HIGH SCHOOL   
453           31R445                             PORT RICHMOND HIGH SCHOOL   
454           31R450                                    CURTIS HIGH SCHOOL   
455           31R455                               TOTTENVILLE HIGH SCHOOL   
456           31R460                           SUSAN E. WAGNER HIGH SCHOOL   
457           31R470                                   CONCORD HIGH SCHOOL   
458           31R600     RALPH R. MCKEE CAREER AND TECHNICAL EDUCATION ...   
459           31R605                   STATEN ISLAND TECHNICAL HIGH SCHOOL   
460           31R607       YOUNG ADULT BOROUGH CENTER AT STATEN IS TECH HS   
461           32K403                  ACADEMY FOR ENVIRONMENTAL LEADERSHIP   
462           32K545           EBC HIGH SCHOOL FOR PUBLIC SERVICE–BUSHWICK   
463           32K549                    BUSHWICK SCHOOL FOR SOCIAL JUSTICE   
464           32K552                             ACADEMY OF URBAN PLANNING   
465           32K554                  ALL CITY LEADERSHIP SECONDARY SCHOOL   
466           32K556     BUSHWICK LEADERS HIGH SCHOOL FOR ACADEMIC EXCE...   
467         

## Question 1

** What could be the future values of CR Average scores based on the CR average scores from SAT Examination 2010?**

### CHI-SQUARE CALCULATION

***Calculating Chi-Square value for CR_Average_Score_2010 and predicted CR_Average_Score value of 2012***

In [73]:
observed = SAT_Result_2010['CR_Average_Score_2010'].mean()
expected = SAT_Result['CR_Average_Score'].mean()
print('Observed value: ', observed)
print('Expected value: ',expected)
chi_squared_stat = (((observed-expected)**2)/expected)
print('Chi-Square value: ',chi_squared_stat)

Observed value:  404.2086956521739
Expected value:  400.7489539748954
Chi-Square value:  0.029868605656417


In [51]:
import scipy
from scipy.stats import chisquare
chisquare(SAT_Result_2010['CR_Average_Score_2010'].mean(),f_exp=SAT_Result['CR_Average_Score'].mean())

Power_divergenceResult(statistic=0.029868605656417, pvalue=nan)

**Interpretation: Here we have calculated the value of Chi square for CR average scores of year 2010 which is considered to be an observed value and future value of CR average is considered to be an expected one.
From the above calculation we got the Chi-Square value:  0.029868605656417 which is too less to be considered and hence it can be concluded that CR average marks of 2012 could be approximately similar to the marks of 2010 and hence there is not a significant difference in the CR average marks of both years**

### LINEAR REGRESSION CALCULATION

***Calculating Coefficients,p-value, R-squared value***

In [52]:
import statsmodels.formula.api as smf
lm= smf.ols(formula='Applicants_2010 ~ CR_Average_Score_2010' , data=SAT_Result_2010).fit()
lm.params # print coefficients

Intercept               -350.904942
CR_Average_Score_2010      1.124313
dtype: float64

***Calculating prediction of CR average score based on the exisiting value from SAT Result 2010***

In [53]:
lm.predict(SAT_Result_2010.CR_Average_Score_2010)

Serial Number  DB_Number_2010
0              01M292             88.701465
1              01M448             92.074404
2              01M450            119.057918
3              01M458             81.955586
4              01M509            103.317535
5              01M515              2.129359
6              01M539            287.704877
7              01M650            111.187726
8              01M696            357.412287
9              02M047            104.441848
10             02M288            123.555170
11             02M294             78.582647
12             02M296             95.447343
13             02M298            125.803796
14             02M300            104.441848
15             02M303             77.458334
16             02M305            111.187726
17             02M308            108.939100
18             02M313            122.430857
19             02M316             99.944595
20             02M374             93.198717
21             02M376            103.317535
22

**Interpretations: Here we are predicting the future values of CR average score**

***Calculating the p-value of CR average score***

In [54]:
lm.pvalues

Intercept                8.625054e-15
CR_Average_Score_2010    3.568955e-23
dtype: float64

***Calculating R-Squared value for CR average score***

In [55]:
lm.rsquared

0.1933709020227159

***t-test for CR average score***

In [83]:
import scipy
from scipy import stats
stats.ttest_rel(SAT_Result_2010['CR_Average_Score_2010'].head(),SAT_Result['CR_Average_Score'].head())

Ttest_relResult(statistic=1.1785696654867619, pvalue=0.30389138776396163)

**Interpretation: From the above t -test calculation , we got a p-value as 0.30389138776396163 which is greater than 0.05. In other words, p> 0.05 then SAT result data sets for year 2010 and 2012 are related to each other and in this case null hypothesis cannot be rejected. Hence there is a difference between the means of CR values and we can not conclude here that significant difference exists.**

## Question 2

**What could be the future values of Math Average scores based on the Math average scores from SAT Examination 2010?**

***Calculating Chi-Square value for Math_Average_Score_2010 and predicted Math_Average_Score value of 2012***

In [74]:
observed = SAT_Result_2010['Math_Average_Score_2010'].mean()
expected = SAT_Result['Math_Average_Score'].mean()
chi_squared_stat = (((observed-expected)**2)/expected)
print('Observed value: ', observed)
print('Expected value: ',expected)
print('Chi-Square value: ',chi_squared_stat)

Observed value:  412.78478260869565
Expected value:  413.3242677824268
Chi-Square value:  0.0007041547650642788


In [23]:
chisquare(SAT_Result_2010['Math_Average_Score_2010'].mean(),f_exp=SAT_Result['Math_Average_Score'].mean())

Power_divergenceResult(statistic=0.00070415476506427877, pvalue=nan)

**Interpretation: Here we have calculated the value of Chi square for Math average scores of year 2010 which is considered to be an observed value and future value of Math average is considered to be an expected one.
From the above calculation we got the Chi-Square value:  0.0007041547650642788 which is too less to be considered and hence it can be concluded that Math average marks of 2012 could be approximately similar to the marks of 2010 and hence there is not a significant difference in the Math average marks of both years**

### LINEAR REGRESSION CALCULATION

***Calculating Coefficients,p-value, R-squared value***

In [65]:
import statsmodels.formula.api as smf
lm= smf.ols(formula='Applicants_2010 ~ Math_Average_Score_2010' , data=SAT_Result_2010).fit()
lm.params # print coefficients

Intercept                 -401.375805
Math_Average_Score_2010      1.223223
dtype: float64

***Calculating prediction of CR average score based on the exisiting value from SAT Result 2010***

In [66]:
lm.predict(SAT_Result_2010.Math_Average_Score_2010)

Serial Number  DB_Number_2010
0              01M292            118.494113
1              01M448            111.154773
2              01M450            125.833453
3              01M458             51.216829
4              01M509            102.592210
5              01M515            249.379010
6              01M539            311.763400
7              01M650             89.136753
8              01M696            342.343983
9              02M047            106.261880
10             02M288            102.592210
11             02M294             43.877489
12             02M296            106.261880
13             02M298            146.628250
14             02M300            144.181803
15             02M303             54.886499
16             02M305             95.252869
17             02M308             64.672286
18             02M313            111.154773
19             02M316             98.922540
20             02M374             58.556169
21             02M376            102.592210
22

**Interpretations: Here we are predicting the future values of math average score**

***Calculating the p-value of CR average score***

In [67]:
lm.pvalues

Intercept                  3.574097e-25
Math_Average_Score_2010    2.678473e-37
dtype: float64

***Calculating R-Squared value for CR average score***

In [68]:
lm.rsquared

0.29951028543413316

***t-test for Math average score***

In [84]:
import scipy
from scipy import stats
stats.ttest_rel(SAT_Result_2010['Math_Average_Score_2010'].head(),SAT_Result['Math_Average_Score'].head())

Ttest_relResult(statistic=-0.10341753799900384, pvalue=0.9226091861131176)

**Interpretation: From the above t -test calculation , we got a p-value as 0.9226091861131176 which is greater than 0.05. In other words, p> 0.05 then SAT result data sets for year 2010 and 2012 are related to each other and in this case null hypothesis cannot be rejected. Hence there is a difference between the means of Math average values and we can not conclude here that significant difference exists.**

## Question 3

**What could be the future values of Writing Average scores based on the Writing average scores from SAT Examination 2010?**

***Calculating Chi-Square value for Writing_Average_Score_2010 and predicted Writing_Average_Score value of 2012***

In [75]:
observed = SAT_Result_2010['Writing_Average_Score_2010'].mean()
expected = SAT_Result['Writing_Average_Score'].mean()
chi_squared_stat = (((observed-expected)**2)/expected)
print('Observed value: ', observed)
print('Expected value: ',expected)
print('Chi-Square value: ',chi_squared_stat)

Observed value:  397.5782608695652
Expected value:  393.8682008368201
Chi-Square value:  0.03494708488100381


In [26]:
chisquare(SAT_Result_2010['Writing_Average_Score_2010'].mean(),f_exp=SAT_Result['Writing_Average_Score'].mean())

Power_divergenceResult(statistic=0.034947084881003808, pvalue=nan)

**Interpretation: Here we have calculated the value of Chi square for Writing average scores of year 2010 which is considered to be an observed value and future value of Writing average is considered to be an expected one.
From the above calculation we got the Chi-Square value:  0.03494708488100381 which is too less to be considered and hence it can be concluded that Writing average marks of 2012 could be approximately similar to the marks of 2010 and hence there is not a significant difference in the Writing average marks of both years**

***Calculating Coefficients,p-value for model coefficients, R-squared value***

In [63]:
import statsmodels.formula.api as smf
lm= smf.ols(formula='Applicants_2010 ~ Writing_Average_Score_2010' , data=SAT_Result_2010).fit()
lm.params # print coefficients

Intercept                    -347.373416
Writing_Average_Score_2010      1.134181
dtype: float64

***Calculating prediction of Writing average score based on the exisiting value from SAT Result 2010***

In [64]:
lm.predict(SAT_Result_2010.Writing_Average_Score_2010)**Interpretations: Here we are predicting the future values of math average score**

Serial Number  DB_Number_2010
0              01M292             89.286153
1              01M448             91.554515
2              01M450            108.567225
3              01M458             81.346888
4              01M509            102.896322
5              01M515              8.759324
6              01M539            296.841221
7              01M650            107.433044
8              01M696            367.160424
9              02M047             89.286153
10             02M288            114.238129
11             02M294             67.736720
12             02M296             96.091237
13             02M298            132.385020
14             02M300            123.311574
15             02M303             80.212708
16             02M305            118.774851
17             02M308            119.909032
18             02M313             99.493780
19             02M316             96.091237
20             02M374             90.420334
21             02M376            102.896322
22

**Interpretations: Here we are predicting the future values of writing average score**

***Calculating the p-value of writing average score***

In [48]:
lm.pvalues

Intercept                     1.601534e-15
Writing_Average_Score_2010    1.984800e-24
dtype: float64

***Calculating R-Squared value for Writing average score***

In [49]:
lm.rsquared

0.20339829083510508

***t-test for Writing average score*** 

In [86]:
import scipy
from scipy import stats
stats.ttest_rel(SAT_Result_2010['Writing_Average_Score_2010'].head(),SAT_Result['Writing_Average_Score'].head())

Ttest_relResult(statistic=6.9577401497571989, pvalue=0.0022424258812041289)

**Interpretation: From the above t -test calculation , we got a p-value as 0.0022424258812041289 which is less than 0.05. In other words, p < 0.05 then hence in this case null hypothesis can be rejected. Hence there is not any difference between the means of Writing average values and we can conclude here that significant difference exists.**

*** References: ***

***https://opendata.cityofnewyork.us/data/#datasetscategory***

***https://data.cityofnewyork.us/Education/SAT-Results/f9bf-2cp4***

***http://www.sportsci.org/resource/stats/pvalues.html***